### ISSUE TITLE
Don't report reportUnannotatedClassAttribute for class `__slots__` attribute

### ISSUE THREAD

**Issue by ashb** [DetachHead/basedpyright#860]
For example:
```python
class TaskInstanceOperations:
    __slots__ = ["client"]
```

This issues a report of

> 1. Type annotation for attribute `__slots__` is required because this class is not decorated with `@final` [reportUnannotatedClassAttribute]

I don't think I've ever seen __slots__ annotated before, and I don't think I should have to -- it is special in python and must be Iterable[strings] else it will yield a type error.

**Comment by DetachHead** [comment]
i considered that but then i realised `__slots__` aren’t even checked to make sure they’re iterable let alone a valid subtype so i decided to just leave it for now until i get around to addressing #181
